### ISSUE TITLE
Make non-key, non @mandatory fields optional attributes

### ISSUE THREAD

**Issue by mattreddy3** [mrbandler/cds2types#5]
Another one for whenever you are working on the library again - would be nice to have all fields be optional except for `key` fields and those with the `@mandatory` annotation

**Comment by mrbandler** [comment]
Hi @mattreddy3,

I was always torn on which CDS basis I create optional fields or not.
As I am currently planning a re-write, which will include a configuration file for further customizations of the generated Typescript code, I will add it there. I hope you can live with the current state of the project until I finished the rewrite.

**Comment by mattreddy3** [comment]
Yes, for sure good for me as-is. Just thought it might be a good addition.
 thanks!

p.s. I would love to help but as I am just getting a handle on typescript as it is, I can't do a whole lot outside of give feedback as a user. That said, let me know if you need anything!
